lib/stackprof/report.rb

```diff
@@ -9,7 +9,7 @@ def initialize(data)
     attr_reader :data
 
     def frames(sort_by_total=false)
-      Hash[ *@data[:frames].sort_by{ |iseq, stats| -stats[sort_by_total ? :total_samples : :samples] }.flatten(1) ]
+      *@data[:frames].sort_by{ |iseq, stats| -stats[sort_by_total ? :total_samples : :samples] }.inject({}){|h, (k, v)| h[k] = v; h}
```

Comment:
In Ruby 2.2.3 this code works (while it takes a few minutes and takes 7 GB of memory):
```ruby
Hash[60_000_000.times.map{ |n| [n, n] }].size
 => 60000000
```
Perhaps it's the `*` that makes the difference? I'm not familiar with the data structure, so I'm not sure why the destructuring is needed.